### ISSUE TITLE
OCA\Notes\AppInfo\Application::OCA\Notes\AppInfo{closure}(): Argument #1 ($event) must be of type OCP\EventDispatcher\GenericEvent

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/notes#1152]
```json
{
  "reqId": "QnQGt1hNVsQ9kxPOLJq5",
  "level": 3,
  "time": "2023-11-10T06:40:24+00:00",
  "remoteAddr": "…",
  "user": "…",
  "app": "no app in context",
  "method": "POST",
  "url": "/ocs/v2.php/apps/files_sharing/api/v1/shares?format=json&path=…&shareType=3",
  "message": "OCA\\Notes\\AppInfo\\Application::OCA\\Notes\\AppInfo\\{closure}(): Argument #1 ($event) must be of type OCP\\EventDispatcher\\GenericEvent, OC\\EventDispatcher\\GenericEventWrapper given, called in …/3rdparty/symfony/event-dispatcher/EventDispatcher.php on line 264 in file …/apps/notes/lib/AppInfo/Application.php line 40",
  "userAgent": "GuzzleHttp/6.5.5 PHP/8.1.24",
  "version": "27.1.3.2",
  "exception": {
    "Exception": "Exception",
    "Message": "OCA\\Notes\\AppInfo\\Application::OCA\\Notes\\AppInfo\\{closure}(): Argument #1 ($event) must be of type OCP\\EventDispatcher\\GenericEvent, OC\\EventDispatcher\\GenericEventWrapper given, called in …/3rdparty/symfony/event-dispatcher/EventDispatcher.php on line 264 in file …/apps/notes/lib/AppInfo/Application.php line 40",
    "Code": 0,
    "Trace": [
      {
        "file": "…/lib/private/AppFramework/App.php",
        "line": 183,
        "function": "dispatch",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->",
        "args": [
          [
            "OCA\\Files_Sharing\\Controller\\ShareAPIController"
          ],
          "createShare"
        ]
      },
      {
        "file": "…/lib/private/Route/Router.php",
        "line": 315,
        "function": "main",
        "class": "OC\\AppFramework\\App",
        "type": "::",
        "args": [
          "OCA\\Files_Sharing\\Controller\\ShareAPIController",
          "createShare",
          [
            "OC\\AppFramework\\DependencyInjection\\DIContainer"
          ],
          [
            "ocs.files_sharing.ShareAPI.createShare"
          ]
        ]
      },
      {
        "file": "…/ocs/v1.php",
        "line": 64,
        "function": "match",
        "class": "OC\\Route\\Router",
        "type": "->",
        "args": [
          "/ocsapp/apps/files_sharing/api/v1/shares"
        ]
      },
      {
        "file": "…/ocs/v2.php",
        "line": 23,
        "args": [
          "…/ocs/v1.php"
        ],
        "function": "require_once"
      }
    ],
    "File": "…/lib/private/AppFramework/Http/Dispatcher.php",
    "Line": 169,
    "Previous": {
      "Exception": "TypeError",
      "Message": "OCA\\Notes\\AppInfo\\Application::OCA\\Notes\\AppInfo\\{closure}(): Argument #1 ($event) must be of type OCP\\EventDispatcher\\GenericEvent, OC\\EventDispatcher\\GenericEventWrapper given, called in …/3rdparty/symfony/event-dispatcher/EventDispatcher.php on line 264",
      "Code": 0,
      "Trace": [
        {
          "file": "…/3rdparty/symfony/event-dispatcher/EventDispatcher.php",
          "line": 264,
          "function": "OCA\\Notes\\AppInfo\\{closure}",
          "class": "OCA\\Notes\\AppInfo\\Application",
          "type": "->",
          "args": [
            "*** sensitive parameters replaced ***"
          ]
        },
        {
          "file": "…/3rdparty/symfony/event-dispatcher/EventDispatcher.php",
          "line": 239,
          "function": "doDispatch",
          "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
          "type": "->",
          "args": [
            [
              [
                "Closure"
              ],
              [
                "Closure"
              ],
              [
                "Closure"
              ],
              [
                "Closure"
              ],
              [
                "Closure"
              ]
            ],
            "*** sensitive parameters replaced ***",
            "*** sensitive parameters replaced ***"
          ]
        },
        {
          "file": "…/3rdparty/symfony/event-dispatcher/EventDispatcher.php",
          "line": 73,
          "function": "callListeners",
          "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
          "type": "->",
          "args": [
            [
              [
                "Closure"
              ],
              [
                "Closure"
              ],
              [
                "Closure"
              ],
              [
                "Closure"
              ],
              [
                "Closure"
              ]
            ],
            "*** sensitive parameters replaced ***",
            "*** sensitive parameters replaced ***"
          ]
        },
        {
          "file": "…/lib/private/EventDispatcher/SymfonyAdapter.php",
          "line": 121,
          "function": "dispatch",
          "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
          "type": "->",
          "args": [
            "*** sensitive parameters replaced ***",
            "*** sensitive parameters replaced ***"
          ]
        },
        {
          "file": "…/lib/private/Share20/Manager.php",
          "line": 809,
          "function": "dispatch",
          "class": "OC\\EventDispatcher\\SymfonyAdapter",
          "type": "->",
          "args": [
            "*** sensitive parameters replaced ***",
            [
              "Symfony\\Component\\EventDispatcher\\GenericEvent"
            ]
          ]
        },
        {
          "file": "…/apps/files_sharing/lib/Controller/ShareAPIController.php",
          "line": 834,
          "function": "createShare",
          "class": "OC\\Share20\\Manager",
          "type": "->",
          "args": [
            "*** sensitive parameters replaced ***"
          ]
        },
        {
          "file": "…/lib/private/AppFramework/Http/Dispatcher.php",
          "line": 230,
          "function": "createShare",
          "class": "OCA\\Files_Sharing\\Controller\\ShareAPIController",
          "type": "->",
          "args": [
            "*** sensitive parameters replaced ***"
          ]
        },
        {
          "file": "…/lib/private/AppFramework/Http/Dispatcher.php",
          "line": 137,
          "function": "executeController",
          "class": "OC\\AppFramework\\Http\\Dispatcher",
          "type": "->",
          "args": [
            [
              "OCA\\Files_Sharing\\Controller\\ShareAPIController"
            ],
            "createShare"
          ]
        },
        {
          "file": "…/lib/private/AppFramework/App.php",
          "line": 183,
          "function": "dispatch",
          "class": "OC\\AppFramework\\Http\\Dispatcher",
          "type": "->",
          "args": [
            [
              "OCA\\Files_Sharing\\Controller\\ShareAPIController"
            ],
            "createShare"
          ]
        },
        {
          "file": "…/lib/private/Route/Router.php",
          "line": 315,
          "function": "main",
          "class": "OC\\AppFramework\\App",
          "type": "::",
          "args": [
            "OCA\\Files_Sharing\\Controller\\ShareAPIController",
            "createShare",
            [
              "OC\\AppFramework\\DependencyInjection\\DIContainer"
            ],
            [
              "ocs.files_sharing.ShareAPI.createShare"
            ]
          ]
        },
        {
          "file": "…/ocs/v1.php",
          "line": 64,
          "function": "match",
          "class": "OC\\Route\\Router",
          "type": "->",
          "args": [
            "/ocsapp/apps/files_sharing/api/v1/shares"
          ]
        },
        {
          "file": "…/ocs/v2.php",
          "line": 23,
          "args": [
            "…/ocs/v1.php"
          ],
          "function": "require_once"
        }
      ],
      "File": "…/apps/notes/lib/AppInfo/Application.php",
      "Line": 40
    },
    "CustomMessage": "--"
  }
}
```

**Comment by nickvergessen** [comment]
Patched it locally in our 27.1.3 to:
```php
                        \OCP\Server::get(IEventDispatcher::class)->addListener('OCP\Share::preShare', function ($event) {
                                if (!$event instanceof GenericEvent) {
                                        return;
                                }
```

And then uploading files etc works again

**Comment by juliushaertl** [comment]
Thanks Joas, added to #1153 as this has other related fixes to sharing
